File: src/cn/yiiguxing/plugin/translate/BasicExplain.java
package cn.yiiguxing.plugin.translate;

import com.google.gson.annotations.SerializedName;

import java.util.Arrays;

public class BasicExplain {

	@SerializedName(value = "phonetic")
	private String phonetic;
	@SerializedName(value = "uk-phonetic")
	private String phoneticUK;
	@SerializedName(value = "us-phonetic")
	private String phoneticUS;
	@SerializedName(value = "explains")
	private String[] explains;

	public String getPhonetic() {
		return phonetic;
	}

	public void setPhonetic(String phonetic) {
		this.phonetic = phonetic;
	}

	public String getPhoneticUK() {
		return phoneticUK;
	}

	public void setPhoneticUK(String phoneticUK) {
		this.phoneticUK = phoneticUK;
	}

	public String getPhoneticUS() {
		return phoneticUS;
	}

	public void setPhoneticUS(String phoneticUS) {
		this.phoneticUS = phoneticUS;
	}

	public String[] getExplains() {
		return explains;
	}

	public void setExplains(String[] explains) {
		this.explains = explains;
	}

	@Override
	public String toString() {
		return "BasicExplain [phonetic=" + phonetic + ", phoneticUK="
				+ phoneticUK + ", phoneticUS=" + phoneticUS + ", explains="
				+ Arrays.toString(explains) + "]";
	}

}

File: src/main/kotlin/cn/yiiguxing/plugin/translate/model/BasicExplain.kt
package cn.yiiguxing.plugin.translate.model

import com.google.gson.annotations.SerializedName
import java.util.*

data class BasicExplain(
        @SerializedName(value = "phonetic")
        var phonetic: String? = null,
        @SerializedName(value = "uk-phonetic")
        var phoneticUK: String? = null,
        @SerializedName(value = "us-phonetic")
        var phoneticUS: String? = null,
        @SerializedName(value = "explains")
        var explains: Array<String>? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as BasicExplain

        if (phonetic != other.phonetic) return false
        if (phoneticUK != other.phoneticUK) return false
        if (phoneticUS != other.phoneticUS) return false
        if (!Arrays.equals(explains, other.explains)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = phonetic?.hashCode() ?: 0
        result = 31 * result + (phoneticUK?.hashCode() ?: 0)
        result = 31 * result + (phoneticUS?.hashCode() ?: 0)
        result = 31 * result + (explains?.let { Arrays.hashCode(it) } ?: 0)
        return result
    }
}
